### PR TITLE
on grad computed callback

### PR DESCRIPTION
**Pull request by benjamin-work**
* Refactor unit tests to reduce code duplication
* add `on_grad_computed` method to callbacks
* add `GradientNormClipping` callback
* deprecate old gradient clipping parameters
* update documentation and tests

Adresses #139

### CODE REVIEW

File: skorch/tests/conftest.py [hunk 1]
```diff
@@ -4,7 +4,11 @@
 
 import numpy as np
 import pytest
+from sklearn.datasets import make_classification
+from sklearn.datasets import make_regression
+from sklearn.preprocessing import StandardScaler
 from torch import nn
+F = nn.functional
```
[the remark this answers is not in the document]
Comment (benjamin-work) [reply skorch/tests/conftest.py]: Is there a difference? This is shorter.

**Comment by benjamin-work** [comment]
@ottonemo Please review again.

File: skorch/tests/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ottonemo) [reply skorch/tests/conftest.py]: Consistency?
